libindex/migrations/migration1.go

```diff
@@ -154,5 +144,19 @@ const (
         PRIMARY KEY(repo_id, scanner_id, layer_hash)
     );
     CREATE INDEX IF NOT EXISTS repo_scanartifact_lookup_idx ON repo_scanartifact(layer_hash);
+
+    --- ManifestIndex
+    --- A searchable index of a coalesced manifest's content.
+    --- a package id is required.
+    --- either a dist_id or a repo_id maybe null, but not both
+    CREATE TABLE IF NOT EXISTS manifest_index
+    (
+        id            bigserial PRIMARY KEY,
+        package_id    bigint NOT NULL REFERENCES package (id),
+        dist_id       bigint REFERENCES dist (id),
+        repo_id       bigint REFERENCES repo (id),
+        manifest_hash text REFERENCES manifest (hash)
```

Comment:
The not-null rule could be enforced by a table constraint:
```suggestion
        manifest_hash text REFERENCES manifest (hash),
        CHECK (dist_id IS NOT NULL AND repo_id IS NOT NULL)
```

Reply:
Think we need OR logic here. 

dist_id not nil, repo_id not nil: OK (future packages with both)
dist_id nil, repo_id not nil: OK (python packages)
dist_id not nil, repo_id nil: OK (linux packages)
dist_id nil, repo_id nil: Not OK (shouldn't happen)